src/Core/Domain/Currency/Command/DeleteCurrencyCommand.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Currency\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+
+/**
+ * Class DeleteCurrencyCommand is responsible for deleting Currency.
+ */
+class DeleteCurrencyCommand
+{
+    /**
+     * @var CurrencyId
+     */
+    private $currencyId;
+
+    /**
+     *
+     * @param CurrencyId $currencyId
+     */
+    public function __construct(CurrencyId $currencyId)
```

Comment:
Do we really need this `CurrencyId` object ? I'm sorry I dont see what value it brings.

Reply:
Inside it has integrity validations so I believe its one advantage

```php
 if (!is_numeric($currencyId) || $currencyId <= 0) {
            throw new CurrencyException(
                sprintf('Invalid Currency id: %s', var_export($currencyId, true))
            );
        }
```

another thing in my opinion is that the id value in the object can be more easily changed if for example the id would become string instead of int type.

and maybe @sarjon has more advantages in mind why use object instead of plain id :thinking:

Reply:
OK let's keep it for now 😉